pkg/nsx/services/securitypolicy/builder.go

```diff
@@ -38,18 +39,15 @@ var (
 	Int64  = common.Int64
 )
 
-func (service *SecurityPolicyService) buildSecurityPolicyName(obj *v1alpha1.SecurityPolicy, createdFor string) string {
+func (service *SecurityPolicyService) buildSecurityPolicyName(obj *v1alpha1.SecurityPolicy) string {
 	if IsVPCEnabled(service) {
 		// For VPC scenario, we use obj.Name as the NSX resource display name for both SecurityPolicy and NetworkPolicy.
 		return util.GenerateTruncName(common.MaxNameLength, obj.Name, "", "", "", "")
 	}
-	prefix := common.SecurityPolicyPrefix
-	if createdFor != common.ResourceTypeSecurityPolicy {
-		prefix = common.NetworkPolicyPrefix
-	}
-	// For T1 scenario, we use ns-name as the key resource name for SecurityPolicy, it is to be consistent with the
+	// For T1 scenario, we use ns_name as the key resource name for SecurityPolicy, it is to be consistent with the
 	// previous solutions.
-	return util.GenerateTruncName(common.MaxNameLength, fmt.Sprintf("%s-%s", obj.Namespace, obj.Name), prefix, "", "", "")
+	return util.GenerateTruncName(common.MaxNameLength, util.ConnectStrings(common.ConnectorUnderline, obj.Namespace, obj.Name),
+		common.SecurityPolicyPrefix, "", "", "")
```

Comment:
I feel it is more effecient if we use `strings.Join([]string{obj.Namespace, obj.Name}, common.ConnectorUnderline)` here when the string contents are known and not empty

Reply:
Good advice, it makes sense to me. I changed the code as suggested.